packages/x-charts/src/ChartsLegend/LegendPerItem.tsx

```diff
@@ -196,27 +203,21 @@ export function LegendPerItem(props: LegendPerItemProps) {
   return (
     <NoSsr>
       <ChartsLegendRoot className={classes?.root}>
-        {itemsWithPosition.map(({ id, label, color, positionX, positionY }) => (
-          <g
-            key={id}
-            className={classes?.series}
-            transform={`translate(${gapX + (isRTL ? legendWidth - positionX : positionX)} ${gapY + positionY})`}
-          >
-            <rect
-              className={classes?.mark}
-              x={isRTL ? -itemMarkWidth : 0}
-              y={-itemMarkHeight / 2}
-              width={itemMarkWidth}
-              height={itemMarkHeight}
-              fill={color}
-            />
-            <ChartsText
-              style={labelStyle}
-              text={label}
-              x={(isRTL ? -1 : 1) * (itemMarkWidth + markGap)}
-              y={0}
-            />
-          </g>
+        {itemsWithPosition.map((item, i) => (
+          <ChartsLegendItem
+            {...item}
+            key={item.id}
+            index={i}
+            gapX={gapX}
+            gapY={gapY}
+            legendWidth={legendWidth}
+            itemMarkHeight={itemMarkHeight}
+            itemMarkWidth={itemMarkWidth}
+            markGap={markGap}
+            labelStyle={labelStyle}
+            classes={classes}
+            onClick={onItemClick}
```

Comment:
What about delegating the context creation to the parent, such that `LegendPerItem` and `ChartsLegendItem` are agnostic from the king of legend they render

```suggestion
            onClick={onItemClick && (event) => onItemClick(event, item.context)}
```

or `onItemClick = (event, index) => void` and the parent can construct the context base on, the `index``